packages/ui/components/combobox/src/LionCombobox.js

```diff
@@ -480,6 +549,17 @@ export class LionCombobox extends LocalizeMixin(OverlayMixin(LionListbox)) {
     if (typeof this._selectionDisplayNode?.onComboboxElementUpdated === 'function') {
       this._selectionDisplayNode.onComboboxElementUpdated(changedProperties);
     }
+
+    if (changedProperties.has('requireOptionMatch') || changedProperties.has('multipleChoice')) {
+      if (!this.requireOptionMatch && this.multipleChoice) {
+        // TODO implement !requireOptionMatch and multipleChoice flow
+
+        // eslint-disable-next-line no-console
+        console.warn(
+          "multipleChoice and requireOptionMatch=false can't be used at the same time (yet).",
+        );
```

Comment:
```suggestion
        throw new Error(
          "multipleChoice and requireOptionMatch=false can't be used at the same time (yet).",
        );
```

I think throwing an error makes more sense, since you enter an illegal (for now) mode and a consuming dev might not check the console for this.